datafusion/sql/src/unparser/expr.rs

```diff
@@ -2512,4 +2539,36 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_window_func_support_window_frame() -> Result<()> {
+        let default_dialect: Arc<dyn Dialect> =
+            Arc::new(CustomDialectBuilder::new().build());
+
+        let test_dialect: Arc<dyn Dialect> = Arc::new(
+            CustomDialectBuilder::new()
+                .with_window_func_support_window_frame(false)
+                .build(),
+        );
+
+        for (dialect, expected) in [
+            (
+                default_dialect,
+                "rank(a) OVER (ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)",
+            ),
+            (test_dialect, "rank(a) OVER ()"),
```

Comment:
This looks weird. Are we sure this is the correct behavior that we want?

Reply:
This is the expected behavior for Dremio. The rank() function in dremio doesn't support window frame, so when the start_bound and end_bound indicates there's no frame specified in the original query, the frame `ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING` should be omitted in unparsing. And the behavior for `window_func_support_window_frame` method is to unparse without window frame when it returns false. 

I updated the `window_func_support_window_frame` to take the `func_name`, `start_bound`, `end_bound` as input instead of just `func_name`, since the combination of `func_name`, `start_bound`, `end_bound`  to determine whether a window frame should be omitted in unparsing might differ from engine to engine.